Makefile.am

```diff
@@ -141,8 +142,8 @@ libbacktrace_elf_for_test_la_LIBADD = $(BACKTRACE_FILE) elf_for_test.lo \
 	$(VIEW_FILE) $(ALLOC_FILE)
 
 elf_for_test.c: elf.c
-	SEARCH='^#define SYSTEM_BUILD_ID_DIR.*$$'; \
-	REPLACE="#define SYSTEM_BUILD_ID_DIR \"$(TEST_BUILD_ID_DIR)\""; \
+	SEARCH='^#define BUILD_ID_DIR.*$$'; \
+	REPLACE='\0\n#undef SYSTEM_DEBUG_DIR\n#define SYSTEM_DEBUG_DIR "$(TEST_DEBUG_DIR)"'; \
```

Comment:
Ideally, we would be able to just use `-D` flag here but I tried the following

```
--- a/Makefile.am
+++ b/Makefile.am
@@ -144,12 +144,9 @@ libbacktrace_elf_for_test_la_LIBADD = $(BACKTRACE_FILE) elf_for_test.lo \
 	$(VIEW_FILE) $(ALLOC_FILE)
 
 elf_for_test.c: elf.c
-	SEARCH='^#define BUILD_ID_DIR.*$$'; \
-	REPLACE='\0\n#undef SYSTEM_DEBUG_DIR\n#define SYSTEM_DEBUG_DIR "$(TEST_DEBUG_DIR)"'; \
-	$(SED) "s%$$SEARCH%$$REPLACE%" \
-		$< \
-		> $@.tmp
-	mv $@.tmp $@
+	cp $< $@
+
+libbacktrace_elf_for_test_la_CFLAGS = -USYSTEM_DEBUG_DIR -DSYSTEM_DEBUG_DIR=\"$(TEST_DEBUG_DIR)\"
 
 endif HAVE_OBJCOPY_DEBUGLINK
 endif HAVE_ELF
```

without success and autotools are just too confusing to me.

`libbacktrace_elf_for_test_la_CFLAGS` does not seem to affect the compilation of `elf_for_test.lo` (it still gets `-DSYSTEM_DEBUG_DIR` from `AM_CFLAGS`) and it only appears when linking the objects into `libbacktrace_elf_for_test.la`:

<pre>
cp elf.c elf_for_test.c
/nix/store/4xw8n979xpivdc46a9ndcvyhwgif00hz-bash-5.1-p16/bin/bash ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I.    -funwind-tables -frandom-seed=elf_for_test.lo -W -Wall -Wwrite-strings -Wstrict-prototypes -Wmissing-prototypes -Wold-style-definition -Wmissing-format-attribute -Wcast-qual  <strong>-DSYSTEM_DEBUG_DIR=\"/nix/store/dkpyi2cxvjxk74l2pjighshbzncy2hmy-libbacktrace-unstable-2022-12-16/lib/debug\"</strong> -g -O2 -c -o elf_for_test.lo elf_for_test.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -funwind-tables -frandom-seed=elf_for_test.lo -W -Wall -Wwrite-strings -Wstrict-prototypes -Wmissing-prototypes -Wold-style-definition -Wmissing-format-attribute -Wcast-qual -DSYSTEM_DEBUG_DIR=\"/nix/store/dkpyi2cxvjxk74l2pjighshbzncy2hmy-libbacktrace-unstable-2022-12-16/lib/debug\" -g -O2 -c elf_for_test.c  -fPIC -DPIC -o .libs/elf_for_test.o
/nix/store/4xw8n979xpivdc46a9ndcvyhwgif00hz-bash-5.1-p16/bin/bash ./libtool  --tag=CC   --mode=link gcc <strong>-USYSTEM_DEBUG_DIR -DSYSTEM_DEBUG_DIR=\"/build/libbacktrace/usr/lib/debug\"</strong> -g -O2   -o libbacktrace_elf_for_test.la  libbacktrace_elf_for_test_la-atomic.lo libbacktrace_elf_for_test_la-dwarf.lo libbacktrace_elf_for_test_la-fileline.lo libbacktrace_elf_for_test_la-posix.lo libbacktrace_elf_for_test_la-print.lo libbacktrace_elf_for_test_la-sort.lo libbacktrace_elf_for_test_la-state.lo backtrace.lo simple.lo elf_for_test.lo mmapio.lo mmap.lo 
libtool: link: ar cr .libs/libbacktrace_elf_for_test.a .libs/libbacktrace_elf_for_test_la-atomic.o .libs/libbacktrace_elf_for_test_la-dwarf.o .libs/libbacktrace_elf_for_test_la-fileline.o .libs/libbacktrace_elf_for_test_la-posix.o .libs/libbacktrace_elf_for_test_la-print.o .libs/libbacktrace_elf_for_test_la-sort.o .libs/libbacktrace_elf_for_test_la-state.o .libs/backtrace.o .libs/simple.o .libs/elf_for_test.o .libs/mmapio.o .libs/mmap.o 
libtool: link: ranlib .libs/libbacktrace_elf_for_test.a
</pre>

```patch

```